23964.rb

```diff
@@ -0,0 +1,54 @@
+begin
+  require 'bundler/inline'
+rescue LoadError => e
+  $stderr.puts 'Bundler version 1.10 or later is required. Please update your Bundler'
+  raise e
+end
+
+gemfile(true) do
+  source 'https://rubygems.org'
+  gem 'rails', path: "."
+end
+
+require 'action_controller/railtie'
+
+class TestApp < Rails::Application
+  config.root = File.dirname(__FILE__)
+  config.session_store :cookie_store, key: 'cookie_store_key'
+  secrets.secret_token    = 'secret_token'
+  secrets.secret_key_base = 'secret_key_base'
+
+  config.logger = Logger.new($stdout)
+  Rails.logger  = config.logger
+
+  routes.draw do
+    get '/' => 'test#index'
+  end
+end
+
+class TestController < ActionController::Base
+  include Rails.application.routes.url_helpers
+
+  def index
+    etag = response.body
+    render plain:  response["ETag"] = etag
+  end
+end
+
+require 'minitest/autorun'
+require 'rack/test'
+
+class BugTest < Minitest::Test
+  include Rack::Test::Methods
+
+  def test_returns_success
+    get '/'
+
+    assert last_response.headers.has_key?("ETag")
+  end
+
+  private
+    def app
+      Rails.application
+    end
+end
```

Comment:
Looks like you committed this accidentally :smile: I put my scripts in a folder and tell git to ignore it because I would always forget to not commit it.